physics/apsides_test.cpp

```diff
@@ -369,5 +378,364 @@ TEST_F(ApsidesTest, ComputeNodes) {
 
 #endif
 
+// A dedicated fixture for |ComputeCollisionIntervals| because we have many
+// tests for that function.
```

Comment:
Do we have tests covering the cases where the endpoints of the trajectory are under max_radius?

I think there are three cases that are treated specially in the implementation, initial time under max_radius ascending, initial time under max_radius descending, and final time under max_radius descending, but it would probably be safest to test all four combinations.

Reply:
The test that changed has its initial point below `max_radius` and that's a periapsis.

Reply:
Added the other 3 tests.  I also did the TDD thing.